### PR TITLE
OCPBUGS-98868: Fix stale clockClass 6 announcement to downstream ptp4l during T-BC holdover

### PR DESCRIPTION
**Pull request by gtannous-spec**
## Summary
The gmIdentity regex only matched decimal digits, but real PTP clock identities are hex - so every EXTERNAL_GRANDMASTER_PROPERTIES_NP PMC call silently timed out after 2s, every time. That chronic stall gave downstreamAnnounceIWF enough time to push a stale, pre-transition clockClass 6 (LOCKED) to the downstream ptp4l config after the BC had already correctly moved to HOLDOVER and announced clockClass 135.

Fixes the regex to accept hex clock identities, adds a BC-state re-check immediately before each downstream PMC write as a race guard, and adds regression tests against real captured pmc output.

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented outdated timing data from being written after the clock state changes.
  * Improved validation of clock identity and parent data received from timing services.
  * Fixed parsing for valid hexadecimal clock identities, including uppercase values.

* **Tests**
  * Added comprehensive coverage for timing data parsing, validation, updates, and edge cases.

### CODE REVIEW

File: pkg/event/event_tbc.go [hunk 1]
```diff
@@ -407,6 +407,9 @@ func (e *EventHandler) applyIfLockedBC(cfgName, context string, fn func()) bool
 	fn()
 	return true
 }
+func (e *EventHandler) isLockedBC(cfgName, context string) bool {
```
Comment (nocturnalastro) [inline pkg/event/event_tbc.go]: Refactor out an isLocked() then make applyIfLocked Use it

File: pkg/event/event_tbc.go [hunk 2]
```diff
@@ -457,9 +460,18 @@ func (e *EventHandler) downstreamAnnounceIWF(ctx context.Context, cfgName string
 	}
 	glog.Infof("%++v", es)
 	e.announceClockClass(gs.ClockQuality.ClockClass, gs.ClockQuality.ClockAccuracy, cfgName)
+
+	// Re-check we're still LOCKED right before each write: gs/es are a stale snapshot if a slow PMC call let the BC move to HOLDOVER/FREERUN in the meantime.
+	if !e.isLockedBC(cfgName, "before downstream PMC writes") {
+		return
+	}
 	if err := pmc.SetExternalGMPropertiesNP(controlledPortsConfig, es); err != nil {
```
Comment (nocturnalastro) [inline pkg/event/event_tbc.go]: Or use the same pattern as vitaly. 
```go
if !e.applyIfLockedBC(cfgName, "before downstream PMC writes", func(){
      if err = pmc.SetExternalGMPropertiesNP(controlledPortsConfig, es); err != nil {
          glog.Error(err)
      }
}) {
		return
	}
```

File: pkg/protocol/type.go [hunk 3]
```diff
@@ -18,6 +18,21 @@ const (
...
+	keyStepsRemoved = "stepsRemoved"
+)
+
+// clockIdentityPattern matches a PTP clockIdentity as printed by ptp4l/pmc:
+// an 8-octet EUI-64 value formatted as three hex groups (e.g.
+// 507c6f.fffe.1fb16c). Shared by every DataSet field that carries a bare
+// clock identity (ExternalGrandmasterProperties.gmIdentity,
+// ParentDataSet.grandmasterIdentity) so they all stay consistent and
+// correctly reject non-hex/malformed values instead of matching anything.
+const clockIdentityPattern = `[0-9a-fA-F]+\.[0-9a-fA-F]+\.[0-9a-fA-F]+`
```
Comment (nocturnalastro) [inline pkg/protocol/type.go:30]: Would be better if the GM identity stuff was a seperate commit. Then you need less stuff in comments and more in git log.

This should just be more like
```
// clock identities are an 8-octet EUI-64 value formatted as three hex groups (e.g.
// 507c6f.fffe.1fb16c). 
```
explaining why it needs to be its own pattern is superfluous

File: pkg/event/event_tbc.go (hunk 1 above)
Comment (gtannous-spec) [reply pkg/event/event_tbc.go]: Reverted as it was before fixing the nitpick comment !

File: pkg/event/event_tbc.go (hunk 2 above)
Comment (gtannous-spec) [reply pkg/event/event_tbc.go]: Fixed.

File: pkg/protocol/type.go (hunk 3 above)
Comment (gtannous-spec) [reply pkg/protocol/type.go:30]: Done ✅

**Comment by gtannous-spec** [comment]
@nocturnalastro Addressed the comments :) 
let me know if it looks better and/or need any other changes from my side !

**Review by lack — APPROVED** [review]
